ports/rp2/modrp2.c

```diff
@@ -36,11 +41,50 @@
 MP_DECLARE_CONST_FUN_OBJ_VAR_BETWEEN(mod_network_country_obj);
 #endif
 
+// Improved version of
+// https://github.com/raspberrypi/pico-examples/blob/master/picoboard/button/button.c
+STATIC bool __no_inline_not_in_flash_func(bootsel_button)(void) {
+    const uint CS_PIN_INDEX = 1;
+
+    // Disable interrupts and the other core since they might be
+    // executing code from flash and we are about to temporarily
+    // disable flash access.
+    mp_uint_t atomic_state = MICROPY_BEGIN_ATOMIC_SECTION();
+
+    // Set the CS pin to high impedance.
+    hw_write_masked(&ioqspi_hw->io[CS_PIN_INDEX].ctrl,
+        (GPIO_OVERRIDE_LOW << IO_QSPI_GPIO_QSPI_SS_CTRL_OEOVER_LSB),
+        IO_QSPI_GPIO_QSPI_SS_CTRL_OEOVER_BITS);
+
+    // Delay without calling any functions in flash.
+    uint32_t start = timer_hw->timerawl;
+    while ((uint32_t)(timer_hw->timerawl - start) <= MICROPY_HW_BOOTSEL_DELAY_US) {
+        ;
+    }
+
+    bool button_state = !(sio_hw->gpio_hi_in & (1 << CS_PIN_INDEX));
```

Comment:
I think it's worth commenting that this is reading the GPIO HI register which has access to the QSPI IO pins.  Also that pressing the button pulls QSPI_SS low.

Reply:
OK, I added that and a comment for the line below it.